### DISCUSSION TITLE
How to implement cart

### DISCUSSION THREAD

**Discussion by bennjjii** [vendurehq/vendure#940] [Q&A]
Can somebody give me a brief overview of how the cart works for logged in and non logged in users? I am using express + ejs templating, so server needs to render a cart page using the users stored selections. I just don't know where to store this data. Is it a feature of vendure?

Thank you

**Comment by michaelbromley** [comment]
Hi,

The docs covering this are here: https://www.vendure.io/docs/storefront/shop-api-guide/

The cart works the same for both authenticated and guest customers. You execute the `addItemToOrder` mutation and then Vendure will store this in an Order entity, which will be available via the `activeOrder` query.

Note that the correct functioning depends on proper session handling, which is covered here: https://www.vendure.io/docs/storefront/managing-sessions/

**Reply by bennjjii** [reply]
Hi,

That's great, thank you for the pointers
